### PR TITLE
Adding Capabilities support to CreateableStorageKey

### PR DESCRIPTION
**Pull request by alxrsngrtn**
## Summary
- Parity Support with Capability enum in arcs/core/data
- Updated PlanGenerator
- Added plan-generator-tests and Goldens

### CODE REVIEW

File: src/tools/plan-generator.ts [hunk 1]
```diff
@@ -139,6 +146,37 @@ export class PlanGenerator {
     return this.createHandleRegistry.get(handle);
   }
 
+  /** Generates Kotlin `Capabilities` from Capabilities. */
+  createCapabilities(capabilities: Capabilities): string {
+    const ktCapabilities  = [];
+
+    if (capabilities.isEmpty()) {
```
Comment (cromwellian) [inline src/tools/plan-generator.ts:153]: The default is TiedToRuntime if no argument is passed. Won't this break existing code by forcing the second parameter to be empty?

Comment (alxrsngrtn) [reply src/tools/plan-generator.ts:153]: There's code above that checks for this. If the capabilities returned match Empty, then we don't add a second argument. 

in runtime/recipe/handle.ts, the default capability is empty.

File: src/tools/plan-generator.ts [hunk 2]
```diff
@@ -125,11 +126,17 @@ export class PlanGenerator {
       return ktUtils.applyFun('StorageKeyParser.parse', [quote(handle.storageKey.toString())]);
     }
     if (handle.fate === 'create') {
-      return ktUtils.applyFun('CreateableStorageKey', [quote(this.createCreateHandleId(handle))]);
+      const createKeyArgs = [quote(this.createCreateHandleId(handle))];
+      const capabilities = this.createCapabilities(handle.capabilities);
+      if (capabilities !== 'Capabilities.Empty') {
```
Comment (alxrsngrtn) [inline src/tools/plan-generator.ts:131]: @cromwellian This is where the default case gets handled.

**Comment by alxrsngrtn** [comment]
Running manual presubmit check. Will merge if this passes.

**Comment by alxrsngrtn** [comment]
A few adjustments need to be made to support the Queryable capability on the Kotlin side of things. Will wait for #5118 before merging this PR.

File: java/arcs/core/storage/keys/DatabaseStorageKey.kt [hunk 3]
```diff
@@ -95,7 +95,7 @@ sealed class DatabaseStorageKey(
         fun registerKeyCreator() {
             CapabilitiesResolver.registerKeyCreator(
                 DATABASE_DRIVER_PROTOCOL,
-                Capabilities.Persistent
+                Capabilities.PersistentQueryable
```
Comment (alxrsngrtn) [inline java/arcs/core/storage/keys/DatabaseStorageKey.kt:98]: @mmandlis Does this look right?

Comment (mmandlis) [reply java/arcs/core/storage/keys/DatabaseStorageKey.kt:98]: yes

**Pull request merged**